### PR TITLE
Extend External Node Integration test for VPC ID and VM Name.

### PR DESCRIPTION
**Pull request by archanapholla**
- Use VPC ID as match criteria in CES for external node life cycle test.
- Similarly use VM Name as another match criteria.

### CODE REVIEW

File: test/integration/externalnode_test.go [hunk 1]
```diff
@@ -119,58 +163,97 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
+			if vm != "" {
+				output = append(output, vm)
+			}
+		}
+		return output
+	}
+
+	checkEqual := func(actual, expected []string) bool {
+		if len(actual) != len(expected) {
+			logf.Log.Info("the value is not correct", "actual", actual, "expected", expected)
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:178]: is this error or info msg ?

File: test/integration/externalnode_test.go [hunk 2]
```diff
@@ -119,58 +163,97 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
-			logf.Log.Info("Validate External Node Name", "Actual", actualResult, "Expected", expectedResult)
-			Expect(actualResult == expectedResult).To(BeTrue())
-
-			// Validate spec field in the generated External Node CR.
-			spec := &antreav1alpha1.ExternalNodeSpec{
-				Interfaces: []antreav1alpha1.NetworkInterface{
-					{
-						Name: strings.ToLower(nic),
-						IPs:  []string{privateIp, publicIp},
+			logf.Log.Info("Validate generated External Node Name(s) with Expected",
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:214]: Keep less logging. This gets printed on console. Only for important steps and move other to debug

File: test/integration/externalnode_test.go [hunk 3]
```diff
@@ -219,65 +300,81 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
+				for _, expectedEn := range expectedResult {
+					fetchKey := client.ObjectKey{
+						Name:      expectedEn,
+						Namespace: namespace.Name,
+					}
+					// Check if each External Node(matching expectedResult) is getting created.
+					err := checkEnCreated(fetchKey, en)
 					if err != nil {
-						logf.Log.Error(err, "External Entity deletion failed", "fetch key", fetchKey)
+						logf.Log.Error(err, "External Node creation failed", "fetch key", fetchKey)
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:333]: nit: External -- external

File: test/integration/externalnode_test.go [hunk 4]
```diff
@@ -219,65 +300,81 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
 				}
-				Expect(err).NotTo(HaveOccurred())
-
-				// Check if External Node is deleted when transitioned from agented = true.
-				err = checkEnDeleted(fetchKey)
+				// Check if External Nodes are deleted when transitioned from agented = true.
+				err = checkAllEnDeleted()
 				if err != nil {
-					logf.Log.Error(err, "External Node deletion failed", "fetch key", fetchKey)
+					logf.Log.Error(err, "External Node deletion failed", "namespace", namespace.Name)
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:369]: nit: same here

File: test/integration/externalnode_test.go [hunk 5]
```diff
@@ -377,22 +476,41 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
-			expectedResult := fmt.Sprintf("virtualmachine-%s", testVm)
+		func(matchKey string) {
+			var matchValue string
+			var expectedResult []string
+
+			switch matchKey {
+			case vpcIDMatch:
+				matchValue = vpcID
+				for _, vm := range vmList {
+					expectedResult = append(expectedResult, fmt.Sprintf("virtualmachine-%s", vm))
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:487]: make virtualmachine const, we might use vm as prefix later

File: test/integration/externalnode_test.go (hunk 1 above)
Comment (archanapholla) [reply test/integration/externalnode_test.go]: Error logging, corrected it

File: test/integration/externalnode_test.go (hunk 2 above)
Comment (archanapholla) [reply test/integration/externalnode_test.go]: Moved a few to debug logging

File: test/integration/externalnode_test.go (hunk 3 above)
Comment (archanapholla) [reply test/integration/externalnode_test.go]: done

File: test/integration/externalnode_test.go (hunk 5 above)
Comment (archanapholla) [reply test/integration/externalnode_test.go]: added

File: test/integration/externalnode_test.go [hunk 6]
```diff
@@ -219,65 +303,81 @@ var _ = Describe(fmt.Sprintf("%s,%s: ExternalNode", focusAws, focusAzure), func(
...
+		isDelete bool) {
 		if agented == true && isDelete == true {
 			err := deleteCloudEntitySelector()
 			Expect(err).ToNot(HaveOccurred())
 
-			err = checkEnDeleted(fetchKey)
+			err = checkAllEnDeleted()
 			if err != nil {
-				logf.Log.Error(err, "External Node delete failed", "fetch key", fetchKey)
+				logf.Log.Error(err, "external Node delete failed", "namespace", namespace.Name)
```
Comment (reachjainrahul) [inline test/integration/externalnode_test.go:314]: you can make Node as node too

**Pull request merged**